examples/hello-world/src/main.rs

```diff
@@ -0,0 +1,118 @@
+extern crate serde;
+
+use mantle::{Address, Context, Service};
+
+use std::collections::{hash_map::Entry, HashMap};
+
+pub type Result<T> = std::result::Result<T, Error>;
+
+#[derive(Debug, Eq, PartialEq, failure::Fail)]
+pub enum Error {
+    #[fail(display = "Unknown error occured.")]
+    Unknown,
+
+    #[fail(display = "The language requested is not supported at this time.")]
+    UnsupportedLanguage,
+
+    #[fail(display = "The language being added already supports Hello World!")]
+    DuplicateEntry,
+}
+
+#[derive(Service, Default)]
+pub struct HelloWorld {
+    owner: Address,
+    helloworlds: HashMap<String, String>,
+}
+
+impl HelloWorld {
+    /// Constructs a new `HelloWorld` service
+    pub fn new(ctx: &Context) -> Result<Self> {
+        let owner = ctx.sender();
+        let mut helloworlds = HashMap::new();
+        helloworlds.insert("English".to_string(), "Hello World!".to_string());
+        helloworlds.insert("Slovenian".to_string(), "Pozdravljen, svet!".to_string());
+        helloworlds.insert("German".to_string(), "Hello Welt!".to_string());
+        helloworlds.insert("French".to_string(), "Bonjour le monde!".to_string());
+
+        Ok(Self {
+            owner,
+            helloworlds,
+            ..Default::default()
+        })
+    }
+
+    /// Get hello world taking as input the desired language
+    pub fn say_hello(&mut self, _ctx: &Context, language: String) -> Result<String> {
+        let helloworld = match self.helloworlds.entry(language) {
+            Entry::Vacant(_) => return Err(Error::UnsupportedLanguage),
+            Entry::Occupied(oe) => oe.into_mut(),
+        };
+        Ok(helloworld.to_string())
+    }
+
+    /// Add a new language Hello World! pair
+    pub fn add_hello(
+        &mut self,
+        _ctx: &Context,
+        language: String,
+        helloworld: String,
+    ) -> Result<()> {
+        match self.helloworlds.entry(language.clone()) {
```

Comment:
Removing `clone()` yields -
```
   |         -------- move occurs because `language` has type `std::string::String`, which does not implement the `Copy` trait
```